### PR TITLE
Migration guide for eCommerce API

### PR DESCRIPTION
**Pull request by aslaksm**

### CODE REVIEW

File: migration.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+toc_max_heading_level: 5
+pagination_next: null
+pagination_prev: null
+---
+
+END_METADATA -->
+
+# Migration
+
+## Why migrate to ePayment API?
```
Comment (cloveras) [inline migration.md:18]:
```suggestion
## Why migrate to the ePayment API?
```

File: migration.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+:::note
+The ePayment API is backwards compatible with the eCom API. However, the eCom API is _not forwards compatible_ with the ePayment API. This means that **payments initiated with the ePayment API can not be modified or retrieved using the eCom API**.
+:::
+
+Merchants are advised to fully migrate over to the ePayment API. However, it is possible to migrate one endpoint at a time, _provided that the Create Payment endpoint is migrated last (see above note)_.
+
+### Callbacks
+
+For payment callbacks, you no longer have to submit the `CallbackPrefix` as part of the Initiate Payment request. Instead, you can use the [Webhooks API](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/webhooks) to register URLs that will receive callbacks whenever various events occur for your payments.
```
Comment (cloveras) [inline migration.md:44]:
```suggestion
For payment callbacks, you no longer have to submit the `callbackPrefix` as part of the Initiate Payment request. Instead, you can use the [Webhooks API](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/webhooks) to register URLs that will receive callbacks whenever various events occur for your payments.
```

**Review by rebekaburnett — APPROVED** [review]
I don't have time to nitpick, but I'm sure it doesn't need that. :-p

File: migration.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+The Webhooks API provides _Guaranteed Delivery_ callbacks. If the callback is not successfully received on your end, we will retry sending it for several days. In addition, you can now receive callbacks for _all_ adjustments to your payment. Head over to the [Webhooks documentation](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/webhooks) to read more.
```
Comment (cloveras) [inline migration.md:46]:
```suggestion
The Webhooks API provides _guaranteed delivery_ callbacks. If the callback is not successfully received on your end, we will retry sending it for several days. In addition, you can now receive callbacks for _all_ adjustments to your payment. Head over to the [Webhooks documentation](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/webhooks) to read more.
```

File: migration.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+### Payment flows
+
+In the eCom API, merchants could choose between three flows by setting the flags `isApp` and `skipLandingPage`. These flows exist in ePayment as well: Instead of setting the flags, you now simply decide which flow you want through the `userFlow` property. Here's how the fields correspond to each other:
+
+* `isApp: false` and `skipLandingPage: false` -> `WEB_REDIRECT`
+* `isApp: true` and `skipLandingPage: false`  -> `NATIVE_REDIRECT`
+* `skipLandingPage: true`                     -> `PUSH_MESSAGE`
+
+The ePayment API also supports a new flow, [the `QR` flow](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/qr-payments).
```
Comment (cloveras) [inline migration.md:56]: Explain briefly how it works? That there is no roundtrip to "translate" the Dee-link to a QR code, maybe?

File: migration.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+### Renamed and altered fields
+
+* `fallBack` -> Renamed `returnUrl`
+* `scope` -> Moved inside `profile` object
```
Comment (cloveras) [inline migration.md:61]:
```suggestion
* `scope` -> Moved inside the `profile` object
```

File: migration.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
+<!-- START_METADATA
+---
+title: Migration
+sidebar_label: Migration
+id: migration
+sidebar_position: 11
```
Comment (OFollan) [inline migration.md:6]: Where do we want this, after the Checklist maybe?

```suggestion
sidebar_position: 110
```

File: migration.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+description: Guide for existing merchants who wish to migrate their integration to the ePayment API.
+toc_min_heading_level: 2
+toc_max_heading_level: 5
```
Comment (OFollan) [inline migration.md:9]: No idea what these mean 😅

File: migration.md (hunk 3 above)
Comment (aslaksm) [reply migration.md:46]: I figured it's enough to have that covered in the webhooks documentation. We'll probably change what type of events we respond to in the future, and it's easier to keep things updated if we only have to keep track of them in one place.

File: migration.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+### Callbacks
+
+For payment callbacks, you no longer have to submit the `CallbackPrefix` as part of the Initiate Payment request. Instead, you can use the [Webhooks API](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/webhooks) to register URLs that will receive callbacks whenever various events occur for your payments.
+
+The Webhooks API provides _Guaranteed Delivery_ callbacks. If the callback is not successfully received on your end, we will retry sending it for several days. In addition, you can now receive callbacks for _all_ adjustments to your payment. Head over to the [Webhooks documentation](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/webhooks) to read more.
+
+### Payment flows
+
+In the eCom API, merchants could choose between three flows by setting the flags `isApp` and `skipLandingPage`. These flows exist in ePayment as well: Instead of setting the flags, you now simply decide which flow you want through the `userFlow` property. Here's how the fields correspond to each other:
```
[the remark this answers is not in the document]
Comment (aslaksm) [reply migration.md:50]: Missing punctuation in your suggestion

File: migration.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+* `fallBack` -> Renamed `returnUrl`
+* `scope` -> Moved inside `profile` object
+* `transactionText` -> Renamed `paymentDescription`
+* `orderId` -> Renamed `reference`
+* `phoneNumber`: Now requires country code/MSISDN format
+
+
+### Payment Method
+
+The ePayment API supports [Free standing Card Payments](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/free-standing-card-payments). Merchants are required to provide a value to the paymentMethod field to decide if this should be used.
```
Comment (cloveras) [inline migration.md:69]: Is is "free-standing with hyphen, @rebekaburnett? We should avoid capitalization of title words, I think it's in the style guide.

 ```suggestion
The ePayment API supports [free-standing card payments](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/free-standing-card-payments). Merchants are required to provide a value to the paymentMethod field to decide if this should be used.
```

File: migration.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+:::note
+The Wallet PaymentMethod means the user will use the Vipps app to pay. _The Card PaymentMethod is intended for users who don't want to use the app to pay._
```
Comment (cloveras) [inline migration.md:72]:
```suggestion
The `WALLET?` payment method means the user will use the Vipps app to pay.  The `CARD` payment method is intended for users who are not able to (or don't want to) use the app to pay.
```

File: migration.md (hunk 9 above)
Comment (aslaksm) [reply migration.md:69]: I thought it was Free-standing as well, but we don't call it that in the Features article

File: migration.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+:::
+### Express Payments
+
+Express payments on the ePayment API will be available soon.
+
+
+### Direct Capture
+
+Merchants that are configured for Direct Capture now need to set the `directCapture` flag to `true`.
```
Comment (cloveras) [inline migration.md:81]:
```suggestion
Merchants that are configured for direct capture now need to set the `directCapture` flag to `true`.
See [Reserve and capture](https://developer.vippsmobilepay.com/docs/vipps-developers/common-topics/reserve-and-capture).
```

File: migration.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+* [Free Standing Card Payments](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/free-standing-card-payments)
+* [QR Payments](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/qr-payments)
+* [Webhooks](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/webhooks)
+
+
+## Migrating from the eCom API
+
+The ePayment API expands upon the functionality of the eCom API and simplifies the existing flows. Merchants currently using the eCom API should find the ePayment API familiar and intuitive.
+
+The ePayment API is **backwards compatible** with the eCom API. This means that payments initiated via the eCom API can be captured, refunded, cancelled, and retrieved using the ePayment API.
```
Comment (OFollan) [inline migration.md:34]: 🚀

File: migration.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+:::note
+The ePayment API is backwards compatible with the eCom API. However, the eCom API is _not forwards compatible_ with the ePayment API. This means that **payments initiated with the ePayment API can not be modified or retrieved using the eCom API**.
```
Comment (OFollan) [inline migration.md:37]: 🥇

**Review by OFollan — APPROVED** [review]
Fantastic work @aslaksm !

File: migration.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+
+The Webhooks API provides _guaranteed delivery_ callbacks. If the callback is not successfully received on your end, we will retry sending it for several days. In addition, you can now receive callbacks for _all_ adjustments to your payment. Head over to the [Webhooks documentation](https://vippsas.github.io/vipps-developer-docs/docs/APIs/epayment-api/features/webhooks) to read more.
+
+### Payment flows
+
+In the eCom API, merchants could choose between three flows by specifying the parameters `isApp` and `skipLandingPage`. These parameters were added to the original API over the years. The same functionality is available in the ePayment, but smarter: Instead of specifying the parameters, you now simply decide which flow you want through the `userFlow` property. Here's how the fields correspond to each other:
+
+* `isApp: false` and `skipLandingPage: false` -> `WEB_REDIRECT`
+* `isApp: true` and `skipLandingPage: false`  -> `NATIVE_REDIRECT`
```
Comment (tht13) [inline migration.md:51]: @aslaksm native redirect was removed wasn't it?

**Pull request merged**